src/Components/Components/src/Routing/Router.cs

```diff
@@ -93,10 +101,21 @@ public Task SetParametersAsync(ParameterView parameters)
                 throw new InvalidOperationException($"The {nameof(Router)} component requires a value for the parameter {nameof(NotFound)}.");
             }
 
-
             var assemblies = AdditionalAssemblies == null ? new[] { AppAssembly } : new[] { AppAssembly }.Concat(AdditionalAssemblies);
             Routes = RouteTableFactory.Create(assemblies);
+
+            // If we're about to render the router for the first time, then
+            // we need to call the `OnNavigate` handler to ensure that pre-processing
+            // steps are completed before rendering the route. This way, it will work
+            // if you navigate to /PageWithLazyLoadedAssemblies or visit it for the first time.
+            if (OnNavigate != null && !initialOnNavigateCalled)
+            {
+                OnNavigate(NavigationManager.ToBaseRelativePath(_locationAbsolute));
+                initialOnNavigateCalled = true;
```

Comment:
I can't seem to find where this is being used

Reply:
I added an explanation for its use in the comment above. It's explicitly used in L111.